packages/overlays/src/OverlayController.js

```diff
@@ -0,0 +1,192 @@
+import { fakeExtendsEventTarget } from './utils/fake-extends-target.js';
+import { GlobalOverlayController } from './GlobalOverlayController.js';
+import { LocalOverlayController } from './LocalOverlayController.js';
+
+// const loadGlobalOverlayController = async() => import('./GlobalOverlayController.js');
+// const loadLocalOverlayController = async() => import('./LocalOverlayController.js');
+
+/**
+ * @see https://popper.js.org/
+ * @typedef {object} PopperConfig
+ * @property {'auto'|'top'|'right'|'bottom'|'left'|'top-start'|'top-end'|'right-start'|'right-end'|'bottom-start'|'bottom-end'|'left-start'|'left-end'} [placement='bottom']
+ * @property {boolean} [positionFixed=false]
+ * @property {boolean} [eventsEnabled=true]
+ * @property {boolean} [removeOnDestroy=false]
+ * @property {object} [modifiers]
+ * @property {Function} [onCreate]
+ * @property {Function} [onUpdate]
+ */
+
+export class OverlayController {
+  /**
+   * @typedef {object} OverlayConfig
+   * @property {HTMLElement} [elementToFocusAfterHide=document.body] - the element that should be
+   * called `.focus()` on after dialog closes
+   * @property {boolean} [hasBackdrop=false] - whether it should have a backdrop (currently
+   * exclusive to globalOverlayController)
+   * @property {boolean} [isBlocking=false] - hides other overlays when mutiple are opened
+   * (currently exclusive to globalOverlayController)
+   * @property {boolean} [preventsScroll=false] - prevents scrolling body content when overlay
+   * opened (currently exclusive to globalOverlayController)
+   * @property {boolean} [trapsKeyboardFocus=false] - rotates tab, implicitly set when 'isModal'
+   * @property {boolean} [hidesOnEsc=false] - hides the overlay when pressing [ esc ]
+   * @property {boolean} [hidesOnOutsideClick=false] - hides the overlay when clicking next to it,
+   * exluding invoker. (currently exclusive to localOverlayController)
+   * https://github.com/ing-bank/lion/pull/61
+   * @property {TemplateResult} [contentTemplate]
+   * @property {TemplateResult} [invokerTemplate] (currently exclusive to LocalOverlayController)
+   * @property {HTMLElement} invokerNode
+   * @property {HTMLElement} contentNode
+   * @property {boolean} [isModal=false] - sets aria-modal and/or aria-hidden="true" on siblings
+   * @property {boolean} [isGlobal=false] - determines the connection point in DOM (body vs next
```

Comment:
`renderToBody` sounds way better 👍

Reply:
or even better?
```js
type="global" // ['global', 'local']
```

Reply:
Yeah. Problem with global for me that I don't know if it means:
- render to body?
- position relative to viewport?
- both?

We could also introduce smth like the term 'portal' (think other libs use this term and might have seen it in a spec proposal before as well if I'm right)

Reply:
In our current system 'global' means both. But I think we should allow local overlays to be rendered to the body as well. Here an example of a use case that needs this: https://github.com/Hacker0x01/react-datepicker/issues/1366

For me it would make sense if it works like:
- renderToBody: renders to body
- viewportPlacement: determines whether it should position relative to viewport